src/components/UserProfile.js

```diff
@@ -1,11 +1,228 @@
+import { useHistory } from 'react-router-dom'
+import { useState } from 'react'
+
+const UserProfile = ({ usersDb, onUpdateUser, onDeleteUser }) => {
+    let history = useHistory()
+    const [username, setUsername] = useState('')
+    const [password, setPassword] = useState('')
+    const [first_name, setFirstName] = useState('')
+    const [last_name, setLastName] = useState('')
+    const [email, setEmail] = useState('')
+    const [admin, setAdmin] = useState(false)
+    const [showEditFields, setShowEditShowFields] = useState(false) 
+    const [showEditButton, setShowEditButton] = useState(true) 
+    const [showEditPassword, setShowEditPassword] = useState(false)  
+    const [showEditPasswordButton, setShowEditPasswordButton] = useState(true)  
+    const [errorUserName, setErrorUsername] = useState(false)
+    const [errorInputBlank, setErrorInput] = useState(false)
+
+    for (let i = 0; i < usersDb.length; i++) {
+        var displayUsername = usersDb[i].username
+        var displayPassword = usersDb[i].password
+        var displayFirstName = usersDb[i].first_name
+        var displayLastName = usersDb[i].last_name
+        var displayEmail = usersDb[i].email
+    }
```

Comment:
What do these lines do? You are iterating usersDb and assigning them to variables in that for-loop, which doesn't seem to do/affect anything outside of the for-loop?